lib/src/license_detection/license.dart

```diff
@@ -31,6 +33,46 @@ class License {
   }
 }
 
+@sealed
+class Ngram {
+  /// Text for which the hash value was generated.
+  final String text;
```

Comment:
Can't the text always be derived from the set tokens?

What if we just had a list of a tokens? Just curious?

Reply:
like... do we need `text` for anything other than debugging?

If not, then maybe we can just make a derived property...

Note: maybe do that refactoring now... consider this more of a vague feedback :)